PizzaBuilder/PizzaBuilder/Program.cs

```diff
@@ -0,0 +1,173 @@
+
+        public void buildSauce()
+        {
+            this.pizza.add("tomato herb sauce");
+        }
+
+        public void buildToppings()
+        {
+            this.pizza.add("cheese");
+        }
+        public DeepdishPizza getPizza()
+        {
+            DeepdishPizza box = this.pizza;
+            this.Reset();
+            return box;
+        }
+    }
+    public class DeepdishPizza
+    {
+        private List<object> pizzaComponents = new List<object>();
+
+        public void add(string components)
+        {
+            this.pizzaComponents.Add(components);
+        }
+
+        public string makePizza()
+        {
+            string str = "";
+            for (int i = 0; i < this.pizzaComponents.Count; i++)
+            {
+                str += this.pizzaComponents[i] + ",";
+            }
+            return "pizza components:" + str;
+        }
+    }
+    class ThincrustPizzaBuilder : IPizzaBuilder
+    {
+        private ThincrustPizza pizza = new ThincrustPizza();
+
+        public ThincrustPizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+
+        }
+        public void grabPan()
+        {
+            this.pizza.add("flat pan");
+        }
+        public void buildDough()
+        {
+            this.pizza.add("mixed flour");
+        }
+
+        public void buildSauce()
+        {
+            this.pizza.add("cheesy white sauce");
+        }
+
+        public void buildToppings()
+        {
+            this.pizza.add("vegetables");
+        }
+        public ThincrustPizza getPizza()
+        {
+            ThincrustPizza box = this.pizza;
+            this.Reset();
+            return box;
+        }
+    }
+    public class ThincrustPizza
+    {
+            private List<object> pizzaComponents = new List<object>();
+
+            public void add(string components)
+            {
+                this.pizzaComponents.Add(components);
+            }
+
+            public string makePizza()
+            {
+                string str = "";
+                for (int i = 0; i < this.pizzaComponents.Count; i++)
+                {
+                    str += this.pizzaComponents[i] + ",";
+                }
+                return "pizza components:" + str;
+            }
+        }
+
+        class Waiter
+    {
+        private IPizzaBuilder builder;
+        public IPizzaBuilder Builder
+        {
+            set { this.builder = value; }
+        }
+        public void makePizza()
+        {
+            this.builder.buildDough();
+
+            this.builder.grabPan();
+           
+            this.builder.buildSauce();
+
+            this.builder.buildToppings();
+        }
+    }
+    class Program
+    {
+        static void Main(string[] args)
+        {
+                Waiter waiter = new Waiter();
+
+            //First chef is tasked with making thin crust pizzas
+                ThincrustPizzaBuilder chef = new ThincrustPizzaBuilder();
```

Comment:
It would be better if you used polymorphism to define builders (`chef`) , without creating new builder variable for each one.